src/track/beats.h

```diff
@@ -1,49 +1,78 @@
 #pragma once
 
-#include <track/timesignature.h>
-#include <QObject>
-#include <QString>
-#include <QList>
 #include <QByteArray>
+#include <QList>
+#include <QObject>
 #include <QSharedPointer>
+#include <QString>
 
+#include "proto/beats.pb.h"
+#include "util/logger.h"
 #include "util/memory.h"
+#include "util/types.h"
 
-namespace {
-    double kMaxBpm = 500;
-}
-
+namespace mixxx {
 class Beats;
-typedef QSharedPointer<Beats> BeatsPointer;
+using BeatsPointer = std::shared_ptr<Beats>;
+using BeatList = QList<track::io::Beat>;
+using SamplePos = double;
+using FrameNum = double;
+} // namespace mixxx
 
-class BeatIterator {
-  public:
-    virtual ~BeatIterator() {}
-    virtual bool hasNext() const = 0;
-    virtual double next() = 0;
-    virtual bool isBar() const = 0;
-    virtual bool isPhrase() const = 0;
-    virtual void makeBeat() = 0;
-    virtual void makeBar() = 0;
-};
+#define BEAT_MAP_VERSION "BeatMap-1.0"
+// TODO(JVC) This two are deprecated
+#define BEAT_GRID_1_VERSION "BeatGrid-1.0"
+#define BEAT_GRID_2_VERSION "BeatGrid-2.0"
```

Comment:
These should be const variables, probably in an anonymous namespace in the cpp file. Abusing preprocessor macros for this is a legacy hack.